File: WebViewApp/app/src/main/java/com/example/webviewapp/presenter/EditRecordPresenter.java
package com.example.webviewapp.presenter;

import com.example.webviewapp.contract.EditRecordContract;
import com.example.webviewapp.data.DataManager;
import com.example.webviewapp.data.Record;

public class EditRecordPresenter implements EditRecordContract.Presenter {
    private final EditRecordContract.View view;

    public EditRecordPresenter(EditRecordContract.View view) {
        this.view = view;
    }

    @Override
    public void checkContent(String title, String details) {
        view.setConfirmButtonVisibility(title != null && details != null);
    }

    @Override
    public Record getData(long primaryKey) {
        return DataManager.get().queryRecordByPrimaryKey(primaryKey);
    }

    @Override
    public void updateRecord(Record record) {
        DataManager.get().updateRecord(record);
    }
}

File: WebViewApp/app/src/main/java/com/example/webviewapp/presenter/EditRecordPresenter.kt
package com.example.webviewapp.presenter

import com.example.webviewapp.contract.EditRecordContract
import com.example.webviewapp.data.DataManager
import com.example.webviewapp.data.Record

class EditRecordPresenter(private val view: EditRecordContract.View) : EditRecordContract.Presenter {
    override fun checkContent(title: String?, details: String?) {
        view.setConfirmButtonVisibility(title != null && details != null)
    }

    override fun getData(primaryKey: Long): Record? {
        return DataManager.get().queryRecordByPrimaryKey(primaryKey)
    }

    override fun updateRecord(record: Record?) {
        DataManager.get().updateRecord(record)
    }

}